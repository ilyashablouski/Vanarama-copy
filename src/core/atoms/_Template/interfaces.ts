import { IBaseProps } from '../../interfaces/base';

export interface ITemplateProps extends IBaseProps {}
