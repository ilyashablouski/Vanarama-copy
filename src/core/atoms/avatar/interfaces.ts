import { IBaseProps } from '../../interfaces/base';
import { TSize } from '../../../types/size';

export interface IAvatarProps extends IBaseProps {
  src?: string;
  size?: TSize;
  altText?: string;
}
