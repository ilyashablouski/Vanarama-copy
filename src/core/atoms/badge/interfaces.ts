import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';
import { TSize } from '../../../types/size';

export interface IBadgeProps extends IBaseProps {
  size?: TSize;
  color?: TColor;
  label?: string;
}
