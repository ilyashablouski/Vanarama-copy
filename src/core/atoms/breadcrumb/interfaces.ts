import { IBaseProps } from '../../interfaces/base';
import { ILink } from '../../interfaces/link';

export interface IBreadcrumbProps extends IBaseProps {
  items?: ILink[];
}
