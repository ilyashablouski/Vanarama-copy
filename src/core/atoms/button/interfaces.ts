import { ButtonHTMLAttributes } from 'react';
import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';
import { TSize } from '../../../types/size';

export interface IButtonProps
  extends IBaseProps,
    ButtonHTMLAttributes<HTMLButtonElement> {
  color?: TColor;
  fill?: 'solid' | 'outline' | 'clear';
  icon?: React.ReactNode;
  iconColor?: TColor;
  iconPosition?: 'before' | 'after';
  label?: string | React.ReactNode;
  round?: boolean;
  size?: TSize;
  withoutDefaultClass?: boolean;
  customCTAColor?: string;
}
