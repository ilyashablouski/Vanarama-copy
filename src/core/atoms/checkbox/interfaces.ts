import { InputHTMLAttributes, ReactElement } from 'react';
import { IBaseProps } from '../../interfaces/base';
import { TColor } from '../../../types/color';

export interface ICheckboxProps
  extends IBaseProps,
    InputHTMLAttributes<HTMLInputElement> {
  /**
   * A color for the checkbox.
   */
  color?: TColor;
  /**
   * A unique ID for the checkbox node - this is required here because the
   * underlying checkbox is hidden and therefore you can only click on the label
   * and the label needs to know which input it corresponds to.
   */
  id: string;
  /**
   * 	Invalid state of the checkbox.
   */
  invalid?: boolean;
  /**
   * A name for the input node.
   */
  label: string | (string | ReactElement)[];
  /**
   * Outlined state of the checkbox.
   */
  outline?: boolean;
  /**
   * A ref to attach to the checkbox.
   */
  ref?: React.Ref<HTMLInputElement>;
}
