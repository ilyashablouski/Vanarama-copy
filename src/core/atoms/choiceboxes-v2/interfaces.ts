import { IBaseProps } from '../../interfaces/base';
import { Nullish } from '../../../types/common';
import { TColor } from '../../../types/color';

export interface IChoiceBoxesV2Props extends IBaseProps {
  boxClassName?: string;
  labelClassName?: string;
  color?: TColor;
  name?: string;
  values: Array<string | number>;
  selectedValues?: Nullish<Array<string | number>>;
  multiSelect?: boolean;
  disabled?: boolean;
  onChange: (values: Array<string | number>) => void;
  idPrefix?: string;
}
