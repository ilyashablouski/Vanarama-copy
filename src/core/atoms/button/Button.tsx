import cx from 'classnames';
import React from 'react';
import Icon from '../icon';
import { IButtonProps } from './interfaces';

const WrappedButton = React.forwardRef<HTMLButtonElement, IButtonProps>(
  function Button(props, ref) {
    const {
      className,
      color = 'secondary',
      dataTestId,
      disabled,
      fill = 'solid',
      icon,
      iconColor,
      iconPosition,
      label,
      round,
      size = 'regular',
      withoutDefaultClass,
      customCTAColor,
      ...rest
    } = props;

    const renderIcon = () => <Icon icon={icon} color={iconColor} />;
    const customStyles = {
      backgroundColor: customCTAColor,
    };
    if (customCTAColor) {
      return (
        // eslint-disable-next-line react/button-has-type
        <button
          {...rest}
          className={cx(
            {
              // eslint-disable-next-line prettier/prettier
              button: !withoutDefaultClass,
            },
            className,
            `-${color} -${size} -${fill}`,
            {
              '-disabled': disabled,
              '-round': round,
            },
          )}
          disabled={disabled}
          data-testid={dataTestId}
          ref={ref}
          style={customStyles}
        >
          <div className="button--inner">
            {icon && iconPosition === 'before' && renderIcon()}
            {label}
            {icon && iconPosition === 'after' && renderIcon()}
          </div>
        </button>
      );
    }

    return (
      // eslint-disable-next-line react/button-has-type
      <button
        {...rest}
        className={cx(
          {
            // eslint-disable-next-line prettier/prettier
            button: !withoutDefaultClass,
          },
          className,
          `-${color} -${size} -${fill}`,
          {
            '-disabled': disabled,
            '-round': round,
          },
        )}
        disabled={disabled}
        data-testid={dataTestId}
        ref={ref}
      >
        <div className="button--inner">
          {icon && iconPosition === 'before' && renderIcon()}
          {label}
          {icon && iconPosition === 'after' && renderIcon()}
        </div>
      </button>
    );
  },
);

export default React.memo(WrappedButton);
