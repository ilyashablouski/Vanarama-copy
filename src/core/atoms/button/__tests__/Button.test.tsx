import React from 'react';
import renderer from 'react-test-renderer';

import { fireEvent, render, screen } from '@testing-library/react';
import Button from '..';
import { IButtonProps } from '../interfaces';

import Add from '../../../assets/icons/Add';

const optionalProps = {
  className: 'class',
  size: 'small' as const,
  color: 'orange' as const,
  fill: 'solid' as const,
  label: 'Label',
  round: true,
  tabIndex: 5,
  type: 'button' as const,
  value: 'value',
};

const disabled = {
  disabled: true,
};

function getComponent(props?: IButtonProps) {
  return renderer.create(<Button {...props} />).toJSON();
}

describe('<Button />', () => {
  it('renders correctly', () => {
    const tree = getComponent();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with optional props', () => {
    const tree = getComponent(optionalProps);
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with an icon before the label', () => {
    const tree = getComponent({
      ...optionalProps,
      icon: <Add />,
      iconPosition: 'before',
    });
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with disabled', () => {
    const tree = getComponent({ ...optionalProps, ...disabled });
    expect(tree).toMatchSnapshot();
  });

  it('should be able to click the button', () => {
    const onClick = jest.fn();
    render(<Button {...optionalProps} onClick={onClick} />);
    fireEvent.click(screen.getByRole('button'));
    expect(onClick).toHaveBeenCalled();
  });
});
