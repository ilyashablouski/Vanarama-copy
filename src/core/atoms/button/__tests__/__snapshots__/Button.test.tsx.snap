// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  className="button -secondary -regular -solid"
>
  <div
    className="button--inner"
  />
</button>
`;

exports[`<Button /> renders correctly with disabled 1`] = `
<button
  className="button class -orange -small -solid -disabled -round"
  disabled={true}
  tabIndex={5}
  type="button"
  value="value"
>
  <div
    className="button--inner"
  >
    Label
  </div>
</button>
`;

exports[`<Button /> renders correctly with optional props 1`] = `
<button
  className="button class -orange -small -solid -round"
  tabIndex={5}
  type="button"
  value="value"
>
  <div
    className="button--inner"
  >
    Label
  </div>
</button>
`;

exports[`<Button /> should render correctly with an icon before the label 1`] = `
<button
  className="button class -orange -small -solid -round"
  tabIndex={5}
  type="button"
  value="value"
>
  <div
    className="button--inner"
  >
    <i
      className="icon"
    >
      <svg
        height="1em"
        viewBox="0 0 512 512"
        width="1em"
      >
        <line
          style={
            Object {
              "fill": "none",
              "stroke": "var(--class-color)",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeWidth": 32,
            }
          }
          x1={256}
          x2={256}
          y1={112}
          y2={400}
        />
        <line
          style={
            Object {
              "fill": "none",
              "stroke": "var(--class-color)",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeWidth": 32,
            }
          }
          x1={400}
          x2={112}
          y1={256}
          y2={256}
        />
      </svg>
    </i>
    Label
  </div>
</button>
`;
